Feature/9/irss upload question

chfsx opened the pull request:
Hi @kergomard !

This PR introduces a migration for files in test questions of type "File Upload".

The PR is divided into the following commits:

- [Add new Stakeholder](https://github.com/ILIAS-eLearning/ILIAS/commit/984171fecca232cc92e98ce94d9f2275176fa05e)
- [Add migration](https://github.com/ILIAS-eLearning/ILIAS/commit/5918de38bfda5890ad9a123ff6830de4ee8fd342): The actual migration of the existing files. This is done using the table `tst_solutions`. Up to now the file name was stored in `value1` and the display name in `value2`. Now the `ResourceIdentification` is stored in `value1` and in `value2` the string `'rid'` to identify which datasets already use an IRSS storage.
- [Implemented Storage and Consumption of Files](https://github.com/ILIAS-eLearning/ILIAS/commit/6d67c868a86772b98f2fd38709018781c6a96c7a): This covers uploading new files and accessing existing files (old location and new in IRSS).
- [Download all submissions](https://github.com/ILIAS-eLearning/ILIAS/commit/f7f4a466df5d445d03a3979f028658f30f92e6d4): Here are still adjustments in downloading all submissions of a question.

[An open issue](https://github.com/ILIAS-eLearning/ILIAS/compare/trunk...srsolutionsag:ILIAS:feature/9/irss-upload-question?expand=1#diff-f8e6a091f181fe83c9f2a451b1f88787b3966704ebee2c9220cb3910939d5d2aR793) is currently known: Since the storage of the test submission takes place in an AtomQuery (ProcessLocker), no changes can be made to the IRSS, i.e. the files could not be effectively deleted. We can discuss the further procedure together.

chfsx committed:
Add new Stakeholder

chfsx committed:
Add migration

chfsx committed:
Implemented Storage and Consumation of Files

chfsx committed:
Download all submissions

chfsx commented:
I'll have a look at the failing Test asap

kergomard commented:
Thank you very much for this improvement @chfsx !

Let us quickly discuss the issue with deleting synchronously.

Just one little thing:
* [x] Could you please fix the failing tests.

Best,
@kergomard

chfsx committed:
[FIX] proper resource deletion

chfsx committed:
[FIX] missing mocks in fileupload question

chfsx commented:
Hi @kergomard 
- I fixed the failing tests by mocking the newly used services
- I implemented the proper deletion of files in the IRSS as discussed via discord

kergomard reviewed (commented):
Thank you very much @chfsx !

This looks fine and I've just two small things:
- We can get rid of [this call here](https://github.com/ILIAS-eLearning/ILIAS/pull/6344), with ILIAS 10 as this is only necessary for files that are not migrated, right? If I'm right: Could you please be so kind and add a corresponding 'todo'-comment. I will then take care of it.
- And now for a real nit-pick: If I'm getting it right [this here](https://github.com/ILIAS-eLearning/ILIAS/pull/6391/commits/6445ef389fb7db8397e739755be97f12758a9e93#diff-f8e6a091f181fe83c9f2a451b1f88787b3966704ebee2c9220cb3910939d5d2aR788-R789) is one if-statement with two conditions. Could you please combine them or is there a reason not to?

Then I will merge.

Thanks again and best,
@kergomard

chfsx committed:
[FIX] combined if statement

chfsx commented:
@kergomard the link of "this call here" leads me to the PR #6344, is that correct? If so, I don't understand which call is meant :-)

I combined the if statement, this was just copy-pasta

kergomard commented:
I should check better, what I link, you are completely right. I meant [this here](https://github.com/ILIAS-eLearning/ILIAS/pull/6391/files#diff-f8e6a091f181fe83c9f2a451b1f88787b3966704ebee2c9220cb3910939d5d2aR840) . Big sorry!

alex40724 commented:
@chfsx  Please have a look at https://mantis.ilias.de/view.php?id=38167 This migration here might run into the same issue. If no files are found nothing is changed and the same step is performed until the end of the migration.

chfsx committed:
added requested todo comment

chfsx commented:
@kergomard  I added the requested todo comment, thanks!

@alex40724 thanks for the hint, in this case this should no happen since we are not using collections here. I will fix the the issue [38167](https://mantis.ilias.de/view.php?id=38167) asap!

kergomard commented:
Thank you very much  @chfsx ! I merge now and look very much forward to start introducing the IRSS everywhere in the Test & Assessment.
